pkg/ingester-rf1/ingester.go

```diff
@@ -590,7 +592,8 @@ func (i *Ingester) doFlushTick() {
 	// TODO: use multiple flush queues if required
 	// Don't write empty segments if there is nothing to write.
 	if currentFlushCtx.segmentWriter.InputSize() > 0 {
-		i.flushQueues[0].Enqueue(currentFlushCtx)
+		i.flushQueues[i.nextFlushQueue].Enqueue(currentFlushCtx)
+		i.nextFlushQueue = (i.nextFlushQueue + 1) % len(i.flushQueues)
```

Comment:
how do you know it's safe to loop around to the beginning of the `flushQueues` again? do we not need to locate data for the same steam/segment next to eachother?